terraform/modules/monitoring/monitoring.tf

```diff
@@ -0,0 +1,181 @@
+variable "gcp_project" {
+  type = string
+}
+
+variable "gcp_region" {
+  type = string
+}
+
+variable "environment" {
+  type = string
+}
+
+resource "kubernetes_namespace" "monitor" {
+  metadata {
+    name = "monitor"
+  }
+}
+
+resource "kubernetes_service" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+  }
+  spec {
+    selector = {
+      app = kubernetes_pod.prometheus.metadata.0.labels.app
+    }
+    port {
+      port        = 9090
+      target_port = 9090
+    }
+
+    type = "LoadBalancer"
+  }
+}
+
+resource "kubernetes_pod" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+    labels = {
+      app = "Prometheus"
+    }
+  }
+
+  spec {
+    service_account_name = kubernetes_service_account.prometheus_account.metadata[0].name
+    container {
+      image = "prom/prometheus:latest"
+      name  = "prometheus"
+      args = [
+        "--config.file", "/config/prometheus.yml",
+        "--storage.tsdb.path", "/data"
+      ]
+      volume_mount {
+        name       = "config-volume"
+        mount_path = "/config"
+      }
+      volume_mount {
+        name       = "storage-volume"
+        mount_path = "/data"
+      }
+    }
+    security_context {
+      fs_group        = 2000
+      run_as_user     = 1000
```

Comment:
Could you include a comment explaining what these values are and why they're needed? I followed your investigation in the `gcp` channel on the ENPA slack and my takeaway was "ooh, that's a bit confusing".